BUILDER_VAGRANT.md

```diff
@@ -0,0 +1,151 @@
+```
+
+You should now have two files in the `.secrets` directory with the exact
+names as shown below:
+
+```
+$ ls .secrets/
+builder-github-app.pem  habitat-env
+```
+
+### Configure the web frontend
+
+Create a file `components/builder-web/habitat.conf.js` from
+`components/builder-web/habitat.conf.sample.js` and change the following
+settings according to your environment:
+
+* `github_client_id`: should be set to the same value as `GITHUB_CLIENT_ID`
+  in the `habitat-env` file.
+* `github_app_id`: should be set to the same value as `GITHUB_APP_ID`
+  in the `habitat-env` file.
+* `github_app_url`: should be set to the public page of your GitHub App.
+
+## Setup
+
+From project root run:
+
+```
+vagrant destroy -f
+vagrant up
+vagrant ssh
+```
+
+Then, in the VM:
+
+```
+sudo su -
+cd /src
+direnv allow .
+make build-bin build-srv
+make bldr-run-no-build
+```
+
+Wait until the web interface fully started. This can take a few minutes:
+
+```
+[...]
+web.1       |  ----------------------------------
+web.1       |        Local: http://localhost:3000
+web.1       |     External: http://10.0.2.15:3000
+web.1       |  ----------------------------------
+web.1       |           UI: http://localhost:3001
+web.1       |  UI External: http://10.0.2.15:3001
+web.1       |  ----------------------------------
+```
+
+Now go to http://localhost:3000/#/pkgs and click Sign-in. You should be
+redirected to GitHub and asked if you allow the GitHub app for you account. The
+name differs between installations and should match what you set above. After
+accepting, you shold be redirected to Habitat Builder (i.e. localhost:3000) and
+be logged-in.
+
+Finally, you should now be able to create an origin, connect a plan from
+GitHub, etc.
+
+## Building a package
+
+To be able to build a package, your Vagrant instance needs to provide the
+required `core/...` packages (since it's an independent Habitat instance).
+
+First, create an origin `core` for that.
+
+Second, for each required package, download it from upstream Habitat Builder
+(or build it yourself, if necessary) and upload the package to your local
+instance. Make sure `$HAB_BLDR_URL` is *not* set to install from upstream.
+Example for `core/hab-backline`, which is always needed:
+
+```
+test -z "$HAB_BLDR_URL" || echo "\$HAB_BLDR_URL is set, unset it first"
+hab pkg install core/hab-backline
+hab pkg upload --url http://localhost:9636/v1 --auth "${HAB_AUTH_TOKEN}" "/hab/cache/artifacts/core-hab-backline-0.40.0-20171128175957-x86_64-linux.hart" --channel stable
+# The package + all dependencies will be uploaded to your *local*
+# core origin
+```
+
+Now, trigger a new build. For a package with no dependencies, above should
+be enough. Otherwise, repeat the process for every package reported
+missing during the build.
+
+If the build fails due to a missing public key, make sure you have both
+a public and a private key in the `/home/krangschnak/.hab/cache/keys/`
+directory, e.g.
+
+```
```

Comment:
How about adding `mkdir --parent /home/krangschnak/.hab/cache/keys`? In our test setup the directory is not there at all.

Reply:
It's there for me, so there seems to be an inconsistency from the provisioning step depending on other factors. I have not yet found the code responsible.

Either way, I think we should rather get rid of this odd setup than add more steps to workaround.